### PR TITLE
Three variables

### PR DESCRIPTION
**Pull request by chisingh**
This change allows grouping by up to 3 variables.

### CODE REVIEW

File: datamations/datamation_groupby.py [hunk 1]
```diff
@@ -59,24 +61,32 @@ def prep_specs_summarize(self, width=300, height=300):
...
-            "labelExpr": "round(datum.label) == 1 ? '" + labels[0] + "' : '"  +  labels[1] + "'",
+            "values": list(range(1, len(groups)+1)),
+            "labelExpr": sublabel if len(self._by) > 2 else "round(datum.label) == 1 ? '" + labels[0] + "' : '"  +  labels[1] + "'",
             "labelAngle": -90
             },
-            "title": self._by[1] if len(self._by) > 1 else x_axis,
+            "title": self._by[-1] if len(self._by) > 1 else x_axis,
             "scale": {
-            "domain": [0.5, 2.5]
+            "domain": [0.5, 3.5 if len(self._by) > 2 else 2.5]
```
Comment (willdebras) [inline datamations/datamation_groupby.py:89]: I believe on the R side of spec generation, we changed this to just always be [0, 3], which would position 2 axes values equidistant from the sides of container and each other. Not super important to change, but fixes some alignment issues at Giorgi's suggestion. You can see x domain in a few places here:

https://github.com/microsoft/datamations/blob/a4213904a70817da77014192c42300a129d1234b/sandbox/custom_animations/custom-animations-mean-R.json#L815

File: datamations/datamation_frame.py [hunk 2]
```diff
@@ -289,27 +399,36 @@ def datamation_sanddance(self):
...
 
-                require(['d3', 'vega', 'vega-util', 'vega-lite', 'vega-embed', 'gemini', 'app', 'utils', 'layout', 'config', 'hack-facet-view'], function(d3, vega, vegaUtil, vegaLite, vegaEmbed, gemini, app, utils, layout, config, hackFacetView) {
+                require(['d3', 'vega', 'vega-util', 'vega-lite', 'vega-embed', 'gemini', 'html2canvas', 'gifshot', 'download2', 'app', 'utils', 'layout', 'config', 'hack-facet-view', 'custom-animations'], function(d3, vega, vegaUtil, vegaLite, vegaEmbed, gemini, html2canvas, gifshot, download2, app, utils, layout, config, hackFacetView, customAnimations) {
                     window.d3 = d3
                     window.vegaEmbed = vegaEmbed
                     window.gemini = gemini
+                    window.html2canvas = html2canvas
+                    window.gifshot = gifshot
+                    window.CustomAnimations = customAnimations
                     window.%s = App("%s", {specs: %s, autoPlay: true});
```
Comment (willdebras) [inline datamations/datamation_frame.py:428]: I think we will need to merge in the changes to this that call `App` as a method of the datamations module, e.g. `datamations.App` and update the `require` call accordingly.

**Review by willdebras — APPROVED** [review]
Looks good to me. Added a small comment on x domain in relation to some re-alignment problems of the faceted labels we previously had.

Will obviously need to fix conflict with Giorgi's newest update to get all of datamations as an node module.

Otherwise, looks good. Happy to further review if you wanted to export python generated specs, but from code/output level review, this looks good.

File: datamations/datamation_groupby.py (hunk 1 above)
Comment (willdebras) [reply datamations/datamation_groupby.py:89]: Update here would also fix a failed test :) https://github.com/microsoft/datamations/issues/145#issuecomment-1083827203

Comment (chisingh) [reply datamations/datamation_groupby.py]: Done. thanks!

**Comment by chisingh** [comment]
Updated the notebook to use latest node package, apart from implementing the 3-variable groupings.

**Pull request merged**